modules/4337/contracts/Safe4337Module.sol

```diff
@@ -212,9 +212,21 @@ contract Safe4337Module is IAccount, HandlerContext, CompatibilityFallbackHandle
     }
 
     /**
-     * @dev Checks if the signatures length is correct and does not contain additional bytes. The function does not
+     * @notice Checks if the signatures length is correct and does not contain additional bytes. The function does not
      * check the integrity of the signature encoding, as this is expected to be checked by the {Safe} implementation
      * of {checkSignatures}.
+     * @dev Safe account has two types of signatures: EOA and Smart Contract signatures. While the EOA signature is
+     * fixed in size, the Smart Contract signature can be of arbitrary length. Safe encodes the Smart Contract
+     * signature length in the signature data. If appropriate length checks are not performed during the signature
+     * verification then a malicious bundler can pad additional bytes to the signatures data and make the account pay
+     * more gas than needed for user operation validation and reach the verificationGasLimit.
+     * `_checkSignaturesLength` function checks for the presence of any padded bytes to the `signature` data.
+     * However, there is an edge case that `_checkSignaturesLength` function cannot detect.
+     * Since the `signature` field in UserOp is not part of the UserOp hash a malicious bundler can manipulate the
```

Comment:
> Since the `signature` field in UserOp is not part of the UserOp hash

To me this is somewhat self-evident. If it were, you wouldn't need `_checkSignatureLength` in the first place. I wouldn't include it here (its a bit confusing IMO).

Reply:
https://github.com/safe-global/safe-modules/pull/471/commits/07f6e216efee71071df28c00b1f1cfb576a99f5d